### ISSUE TITLE
Send postmaster email to sender if email is too big and not processed by Zammad

### ISSUE THREAD

**Issue by martini** [zammad/zammad#2092]
### Infos:

* Used Zammad version: 2.5
* Installation method (source, package, ..): any
* Operating system: any
* Database + version: any
* Elasticsearch version: any
* Browser + version: any
* Ticket-ID: #1031354, #1025102, #1021533, #1029932, #1049679 

### Expected behavior:

* If an email is too big for Zammad (at least for Zammad max email size setting), send an bounce email (like an postmaster email) to the sender.

### Actual behavior:

* Zammad is just skipping too big emails. The emails will stay in the mailbox as untouched.

**Comment by martinvonwittich** [comment]
If this is going to be changed, I would welcome a configuration option for it - in our case, we don't want Zammad to send any bounce mails, and I prefer the current "don't fetch it from IMAP and show a warning to the administrator" approach.

The reason for this is our pretty complex mail setup: we have a main mail server for our company that is the MX for our domain, and all outgoing and incoming mail goes through this server. As soon as we start using Zammad, some of the mail (but not all of it) will be duplicated into separate IMAP accounts for Zammad to fetch. To illustrate it:

E.g. we have an address info@ourcompany.tld. `info` is configured as a group on our main mail server; this group has an archive which stores a copy of every incoming mail, and it has members (e.g. `martin.von.wittich`), and every group member gets a copy in his inbox too. Currently we are still using Kayako as our ticket system, and Kayako gets a copy via `/etc/aliases`:

```
info:        \info,          info@helpdesk.ourcompany.tld
```

In the future, when we migrate from Kayako, the mail will be duplicated similarly to a Zammad IMAP mailbox. It would therefore be very undesirable for us when one of our "backend systems" like Kayako or Zammad suddenly decided to send their own bounce mail, because the decision whether to accept or to reject mail (e.g. due to size constraints) lies with our main mail server, not with the backend systems that only receive copies from the main server. When our main server (that accepts mail up to 50 MB) has already decided to accept a mail, but the mail is too big for Zammad to accept (because Zammad by default accepts up to 10 MB), then this is simply a configuration error on my side. In that case I want Zammad to inform _me_ and not our customer.

TLDR: I like it the way it currently works :)

**Comment by MrGeneration** [comment]
I can understand Martins argumentation.
We should at least add an information to monitoring, as other wise you can't monitor this cleanly!

The option to enable/disable the auto reply is a good idea - I'd see this inside of settings and would ask the admin within the wizard how he'd like it. Reason is, I'd personally prefer Zammad to have the auto reply on by default.

This impacts quite a lot hosted instances as well, most of those users don't use on E-Mail-Servers and their maximum E-Mail size does depend on the chosen package. Plus, monitoring information are missing within hosted setups (because you normally don't need it) that's why a bounce would be  better at that point.

You see, it's a bit complicated ^^"

**Comment by martini** [comment]
It`s implemented for Zammad 3.2 now. 

1) On upgrade installations there will be not other behavior as it is right now. Only too large emails are shown in Zammad monitoring endpoint.

In case you want to have the now behavior, got do admin -> Channels -> E-Mail -> Settings and change "Send postmaster mail if mail too large"

2) In new setups or if "Send postmaster mail if mail too large" is set to true, a email message is sent out to the sender that the mail is to large (how large it was and what the maximum is) and it was not able to deliver.
